lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -25,7 +28,18 @@ function replaceAsString(
   }
   // Look for the old version number
   const searchString = `"${oldValue}"`;
-  const newString = `"${newValue}"`;
+  let newString = `"${newValue}"`;
+
+  if (patchReg.test(oldValue)) {
+    const regex = regEx(`(patch:${depName}@(npm:)?).*#`);
```

Comment:
Can this reuse the same regex pattern as the `.test()`? Otherwise there's a possibility of drift where the `.text()` matches while the `regex.exec()` does not.

Reply:
yes, my mistake, The patchReg is supposed to be the Detection regex, 
the second regex is supposed to be the Replace part regex, thats why i'm using 2
ill change it